chain/network/src/peer_manager/network_state/routing.rs

```diff
@@ -118,4 +125,43 @@ impl NetworkState {
             .await
             .unwrap_or(Ok(()))
     }
+
+    pub(crate) fn find_route(
+        &self,
+        clock: &time::Clock,
+        target: &PeerIdOrHash,
+    ) -> Result<PeerId, FindRouteError> {
+        match target {
+            PeerIdOrHash::PeerId(peer_id) => {
+                self.graph.routing_table.find_route_from_peer_id(peer_id)
+            }
+            PeerIdOrHash::Hash(hash) => self
+                .tier2_route_back
```

Comment:
Just to make sure I understand can you confirm the following is true? I'm trying to understand why do we only consider TIER2 here. 
- Finding route only makes sense for TIER2 because in TIER1 we can directly reach every peer.

Also what's the difference between route and route back here?